py/precommit.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python3
+"""Runs precommit checks on the repository."""
+import argparse
+import os
+import pathlib
+import subprocess
+import sys
+from typing import List  # pylint: disable=unused-imports
+
+
+def main() -> int:
+    """"
+    Main routine
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--overwrite",
+        help=
+        "Overwrites the unformatted source files with the well-formatted code in place. "
+        "If not set, an exception is raised if any of the files do not conform to the style guide.",
+        action='store_true')
+
+    args = parser.parse_args()
+
+    overwrite = bool(args.overwrite)
+
+    repo_root = pathlib.Path(__file__).parent
+
+    print("YAPF'ing...")
+    if overwrite:
+        subprocess.check_call([
+            "yapf", "--in-place", "--style=style.yapf", "--recursive", "tests",
+            "persipubsub", "setup.py", "precommit.py"
+        ],
+                              cwd=repo_root.as_posix())
+    else:
+        subprocess.check_call([
+            "yapf", "--diff", "--style=style.yapf", "--recursive", "tests",
+            "persipubsub", "bin", "setup.py", "precommit.py"
+        ],
+                              cwd=repo_root.as_posix())
+
+    print("Mypy'ing...")
+    subprocess.check_call(["mypy", "--strict", "persipubsub", "tests"],
+                          cwd=repo_root.as_posix())
+
+    print("Isort'ing...")
+    isort_files = []  # type: List[str]
+    for path in (repo_root / "persipubsub").glob("**/*.py"):
+        isort_files.append(path.as_posix())
+    for path in (repo_root / "tests").glob("**/*.py"):
+        isort_files.append(path.as_posix())
+
+    if overwrite:
+        cmd = [
+            "isort", "--balanced", "--multi-line", "4", "--line-width", "80",
+            "--dont-skip", "__init__.py", "--project", "persipubsub"
+        ]
+        cmd.extend(isort_files)
+        subprocess.check_call(cmd)
+    else:
+        cmd = [
```

Comment:
Please reformat manually for readability and add `# yapf: disable/enable` (also elsewhere in this file)